presto-jdbc/src/test/java/com/facebook/presto/jdbc/TestJdbcResultSet.java

```diff
@@ -205,6 +221,16 @@ public void testObjectTypes()
             // TODO this should fail, as there no java.sql.Timestamp representation for TIMESTAMP '1970-01-01 00:14:15.227ó' in America/Bahia_Banderas
             assertEquals(rs.getTimestamp(column), Timestamp.valueOf(LocalDateTime.of(1969, 12, 31, 15, 14, 15, 227_000_000)));
         });
+
+        statement.execute("CREATE TYPE cat.sch.dist AS integer");
+        checkRepresentation("CAST(1 AS cat.sch.dist)", Types.JAVA_OBJECT, (rs, column) -> {
+            assertEquals(rs.getObject(1), 1);
+        });
+
+        statement.execute("CREATE TYPE cat.sch.pair AS (fst integer, snd varchar)");
+        checkRepresentation("CAST((1,'1001') AS cat.sch.pair)", Types.JAVA_OBJECT, (rs, column) -> {
+            assertEquals(rs.getObject(1), Lists.newArrayList(1, "1001"));
+        });
```

Comment:
Not sure if I understand correctly, do we need to add tests like "CREATE TYPE CAT.SCH.PAIR ..." to test the fix here?

Reply:
This test case covers the scenario I mentioned earlier in this PR description. That is when we try to build and get a PrestoResultSet, we pass an uppercase type define string to get a type signature, that would cause exception for user defined type. The corresponding code is in ``PrestoResultSet.getColumnInfo()``.

So I think maybe it's better to add test case for this scenario here.